controllers/auth.js

```diff
@@ -8,27 +7,22 @@ exports.isAuth = (req, res, next) => {
       error: "Access denied"
     });
   }
-  return jwt.verify(req.cookies.jwt, process.env.SECRET, (err) => {
+  jwt.verify(req.cookies.jwt, process.env.SECRET, (err, jwtPayload) => {
     if (err) {
       return res.status(401).json({
         error: "failed to verify!"
       });
-    } else next();
+    } else {
+      res.locals.jwtPayload = jwtPayload;
+      next();
+    }
   });
 };
 
 exports.isAdmin = (req, res, next) => {
-  jwt.verify(req.cookies.jwt, process.env.SECRET, (err, jwtPaload) => {
-    if (err) {
-      return res.status(401).json({
-        error: "failed to verify!"
-      });
-    }
-    if (jwtPaload.admin === false) {
-      return res.status(401).json({
-        error: "Access denied! Admin Only!"
-      });
-    }
-    next();
-  });
+  if (res.locals.jwtPayload && !res.locals.jwtPayload.admin) {
```

Comment:
Just thinking - this logic isn't quite what we want. What will happen if `jwtPayload` doesn't exist on `res.locals`? Currently it will go inside the `else` block and call `next()`, which we don't want!

We want this:

```js
// if there is no jwtPayload OR jwtPayload.admin is false
if (!res.locals.jwtPayload || !res.locals.jwtPayload.admin) {
   // return 401
} else {
  next()
}
```

Reply:
done